did_doc/src/schema/verification_method/verification_method.rs

```diff
@@ -37,9 +38,16 @@ impl VerificationMethod {
         &self.verification_method_type
     }
 
-    pub fn public_key(&self) -> &PublicKeyField {
+    pub fn public_key_field(&self) -> &PublicKeyField {
         &self.public_key
     }
+
+    pub fn public_key(&self) -> Result<Key, DidDocumentBuilderError> {
```

Comment:
Find it kind of code-smellish that this method return DidDocumentBuilderError - as consumer I would be surprised to get this type of error from this method

Reply:
If you are suggesting it should return `PublicKeyError`, that would likely require the client crate implement a conversion to its error type from both `DidDocumentBuilderError` and `PublicKeyError`. This may be usually necessary, as the client crate will likely be using `Key` extensively anyways, but other use-cases may come up where the client crate extracts the `public_key`, does not use it, and just passes it to another crate. One may argue whether that is an anti-pattern, but not unimaginable, so why complicate the API?

If you are suggesting to rename the error type, then I agree, as the crate is called `did_doc` instead of the original `did_doc_builder`.

Reply:
I see... fair enough, I think rename will be nice though. You can do it separately or here, I am hitting approve button. Feel free to merge